modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java

```diff
@@ -362,8 +363,33 @@ public static BigDecimal toBigDecimal(Number value, int precision, int scale) {
         }
     }
 
+    /** Adjusts precision of {@link SqlTypeName#TIME} value. */
+    public static @Nullable Integer toTimeExact(@Nullable Object object, int precision) {
+        if (object == null) {
+            return null;
+        }
+
+        assert object instanceof Integer : object.getClass();
+
+        return toTimeExact((int) object, precision);
```

Comment:
i see that we can avoid unboxing here, isn`t it ?

Reply:
Can you clarify your suggestion?
Are you talking about removing `@Nullable Integer toTimeExact(@Nullable Object object`?